packages/babel-parser/src/parser/lval.js

```diff
@@ -121,11 +137,13 @@ export default class LValParser extends NodeUtils {
 
         node.type = "AssignmentPattern";
         delete node.operator;
+        // AssignmentPattern is not allowed in LHS
```

Comment:
Also this, can't it be `[ [(a)] = [] ] = []`?

Reply:
Addressed in 00a776c.